tutorials/nextjs-app-serverless-functions-sqldb/index.mdx

```diff
@@ -0,0 +1,948 @@
+  paragraph: This page explains how to deploy your fully serverless Next.js application using Serverless Containers and Serverless SQL Database
+dates:
+  validation: 2024-02-29
+  posted: 2024-02-29
+---
+
+This tutorial will guide you through the deployment of a fully serverless Next.js application using a Serverless Container and a Serverless SQL Database in a few minutes.
+
+You can either deploy your application:
+  - [step by step using Scaleway CLI](#deploy-next.js-application-frontend-using-serverless-containers) to understand each detailed action performed, and the resources and subresources required
+  - [using a Terraform template](#deploy-next.js-application-using-terraform-templates) to deploy your application faster and have a ready-to-use Infrastructure as Code.
+
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com/)
+  - You have installed and initialized the [Scaleway CLI](https://www.scaleway.com/en/docs/developer-tools/scaleway-cli/quickstart/)
+  - You have installed [Docker Engine](https://docs.docker.com/engine/install/)
+  - You have installed [Node 18 or newer](https://nodejs.org/en/download)
+</Message>
+
+## Deploy Next.js application frontend using Serverless Containers
+
+### Initializing the project
+
+1. Run the command below in a terminal to export your access key and secret key as environment variables:
+    ```sh
+    export SCW_ACCESS_KEY=$(scw config get access-key)
+    export SCW_SECRET_KEY=$(scw config get secret-key)
+    ```
+
+2. Run the command below to make sure the environment variables are properly set:
+
+    ```sh
+    scw info 
+    ```
+
+    This command displays your access key and secret key as the last two lines. The `ORIGIN` column displays `env (SCW_ACCESS_KEY)` and `env (SCW_SECRET_KEY)`, and not `default profile`.
+      
+    ```bash
+    KEY                      VALUE                                       ORIGIN
+    (...)
+    access_key               SCWF9DETBF829TAFB1TB                        env (SCW_ACCESS_KEY)
+    secret_key               9a1bcf92-xxxx-xxxx-xxxx-xxxxxxxxxxxx        env (SCW_SECRET_KEY)
+    ```
+
+3. Access the folder where you want to store your project repository and run the following command to initialize a blog template application:
+    ```bash
+    npx create-next-app --example blog-starter my-nextjs-blog
+    ```
+
+4. Type `y` when prompted to install the dependencies:
+    ```bash
+    Need to install the following packages:
+    create-next-app@14.0.3
+    Ok to proceed? (y) y
+    ```
+    This will create a folder named `my-nextjs-blog`. 
+    
+5. Access the folder you just created:
+    ```bash
+    cd my-nextjs-blog
+    ```
+
+6. Make sure you can run the application locally:
+    ```bash
+    npm run dev
+    ```
+
+7. Go to [localhost:3000](http://localhost:3000) in a browser. The blog template displays.
```

Comment:
```suggestion
7. Go to `http://localhost:3000` in a browser. The blog template displays.
```

iirc if there is a clickable link that leads to nowhere it will trigger a 404 error when we run our script